Data/ByteString/Internal.hs

```diff
@@ -339,18 +339,18 @@ packChars cs = unsafePackLenChars (List.length cs) cs
  #-}
 
 unsafePackLenBytes :: Int -> [Word8] -> ByteString
-unsafePackLenBytes len xs0 =
-    unsafeCreate len $ \p -> go p xs0
-  where
-    go !_ []     = return ()
-    go !p (x:xs) = poke p x >> go (p `plusPtr` 1) xs
+unsafePackLenBytes len xs =
+    unsafeCreate len $ \p0 -> foldr
+      (\x go p -> poke p x >> go (p `plusPtr` 1))
+      (\_ -> return ()) xs p0
```

Comment:
```suggestion
unsafePackLenBytes len =
    unsafeCreate len . foldr
      (\x go p -> poke p x >> go (p `plusPtr` 1))
      (\_ -> return ())
```
...if I'm not mistaken.

Reply:
I was a bit worried that foldr wouldn't inline because it is not fully saturated that way, but I guess we want the fusion rules to file anyway so that shouldn't be a problem.